lib/matplotlib/backends/backend_agg.py

```diff
@@ -417,25 +417,21 @@ def restore_region(self, region, bbox=None, xy=None):
         return renderer.restore_region(region, bbox, xy)
 
     def draw(self):
-        """
-        Draw the figure using the renderer
-        """
+        """Draw the figure using the renderer."""
         self.renderer = self.get_renderer(cleared=True)
-        # acquire a lock on the shared font cache
-        RendererAgg.lock.acquire()
-
         toolbar = self.toolbar
         try:
             # if toolbar:
             #     toolbar.set_cursor(cursors.WAIT)
-            self.figure.draw(self.renderer)
+            with RendererAgg.lock:
+                self.figure.draw(self.renderer)
             # A GUI class may be need to update a window using this draw, so
             # don't forget to call the superclass.
             super(FigureCanvasAgg, self).draw()
```

Comment:
No need to keep the lock here too?

Reply:
probably safer to keep it indeed (I thought the super class was only going to read (not write) the buffer but you still don't want to expose a partially drawn buffer).